packages/account/src/providers/provider.ts

```diff
@@ -598,39 +598,41 @@ export default class Provider {
     await this.fetchChainAndNodeInfo();
   }
 
+  async refreshCacheFunction(): Promise<{ chain: ChainInfo; nodeInfo: NodeInfo }> {
+    const data = await this.operations.getChainAndNodeInfo();
+
+    const nodeInfo = {
+      maxDepth: bn(data.nodeInfo.maxDepth),
+      maxTx: bn(data.nodeInfo.maxTx),
+      nodeVersion: data.nodeInfo.nodeVersion,
+      utxoValidation: data.nodeInfo.utxoValidation,
+      vmBacktrace: data.nodeInfo.vmBacktrace,
+    };
+
+    Provider.ensureClientVersionIsSupported(nodeInfo);
+    const chain = processGqlChain(data.chain);
+    Provider.chainInfoCache[this.urlWithoutAuth] = chain;
+    Provider.nodeInfoCache[this.urlWithoutAuth] = nodeInfo;
+
+    return { chain, nodeInfo };
+  }
```

Comment:
@maschad It seems this is just a code refactoring. What am I missing?

Reply:
Ok, you are using the `refreshCache` flag. I understood it now

Reply:
Maybe [this](https://github.com/FuelLabs/fuels-ts/pull/3324/files/00fd2ee1f9b6397bb698251774663d1338221fa3#diff-69a6670d0426c74935ed2830ce552e7df2379c2d112a582306e57569fcd17029R1155).

Reply:
Yes, this is just a refactoring. The main change related to the cache is the addition of the `refreshCache` parameter in `fetchChainAndNodeInfo` below.